### PR TITLE
Setup Skeleton for Scala from Java

### PR DESCRIPTION
**Pull request by rjmontgo**
Created LuxReader TensionReader objects as well as TensionData and
LuxData classes to hold data.

This will be updated as I add more items.

### CODE REVIEW

**Comment by rjmontgo** [comment]
Most recent commit setup the Scala build tool and converted major idioms to scala. Basically setting up Lux Reader, Tension Reader and Data Aggregator.

File: logreader/src/main/scala/main.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import scala.io.Source
+
+object logReader extends App {
+  val iterator = Source.fromResource("myapp.log").getLines()
+  var kPaCount = 0 
+  var kPaSum = 0.0
+  var luxCount = 0
+  var luxSum = 0.0
```
Comment (rjmontgo) [inline logreader/src/main/scala/main.scala]: Is there a way to do this without using var?

File: iot/src/main/scala/main.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+object DataAggregator {
+  def apply(): Behavior[Data] = {
+    Behaviors.setup(context => new DataAggregator(context))
+  }
+}
+
+class DataAggregator(context: ActorContext[Data]) extends AbstractBehavior[Data](context) {
+  import LuxReader._
+  import TensionReader._
+  context.spawn(LuxReader(), "LuxReader") ! ReadLux(context.self)
```
Comment (MatthewEdge) [inline iot/src/main/scala/main.scala]: _Might_ be beneficial to use a extended sealed trait if you move the Data trait to each companion object for ADTs:

```
sealed trait Readable extends TensionData with LuxData
```

File: logreader/src/main/scala/main.scala (hunk 1 above)
Comment (rjmontgo) [reply logreader/src/main/scala/main.scala]: Try to implement with foldLeft

Comment (MatthewEdge) [reply logreader/src/main/scala/main.scala]: One option is a foldLeft, something like (assuming List[String]):

```
lines.foldLeft((0, 0.0, 0, 0.0)) {
  case (kPaCount, kPaSum, luxCount, luxSum) => ...
}
```

**Pull request closed without merging**